components/Tree/TreeNodeDetails/TreeNodeDetails.tsx

```diff
@@ -1,31 +1,273 @@
+import classNames from "classnames";
 import Link from "next/link";
-import { FC } from "react";
+import { FC, useState } from "react";
 import { useTree } from "../../../context/tree";
 import { getTreeNodesMap } from "../../../data";
 import { CloseIcon } from "../../../icons/CloseIcon";
+import BioLink from "../../BioLink/BioLink";
+import BioNavItem from "../../BioNavItem/BioNavItem";
 import s from "./TreeNodeDetails.module.css";
 
-const nodesMap = getTreeNodesMap();
+const navigation = [
+  { id: 1, title: "Биография" },
+  { id: 2, title: "Галерея" },
+];
+
+const mapMonth = (month?: number, day?: number) => {
+  switch (month) {
+    case 1:
+      return day === undefined ? "январь" : "января";
+    case 2:
+      return day === undefined ? "февраль" : "февраля";
+    case 3:
+      return day === undefined ? "март" : "марта";
+    case 4:
+      return day === undefined ? "апрель" : "апреля";
+    case 5:
+      return day === undefined ? "май" : "мая";
+    case 6:
+      return day === undefined ? "июнь" : "июня";
+    case 7:
+      return day === undefined ? "июль" : "июля";
+    case 8:
+      return day === undefined ? "август" : "августа";
+    case 9:
+      return day === undefined ? "сентябрь" : "сентября";
+    case 10:
+      return day === undefined ? "октябрь" : "октября";
+    case 11:
+      return day === undefined ? "ноябрь" : "ноября";
+    case 12:
+      return day === undefined ? "декабрь" : "декабря";
+    default:
+      return undefined;
+  }
+};
+
+const getDate = (year?: number, month?: number, day?: number) => {
+  const monthString = mapMonth(month, day);
+  return year
+    ? monthString
+      ? day
+        ? `${day} ${monthString} ${year}`
+        : `${monthString} ${year}`
+      : `${year}`
+    : undefined;
+};
+
+const useTreeNodeDetails = (selectedNodeId?: string) => {
+  if (selectedNodeId === undefined) {
+    return;
+  }
+  const nodesMap = getTreeNodesMap();
+  const selectedNode = nodesMap[selectedNodeId];
+  const parents = selectedNode.parents.map((parent) => {
+    return { id: parent.id, fullName: nodesMap[parent.id].data.fullName };
```

Comment:
Можно получать весь узел из мапы. 

```suggestion
    const children = selectedNode.children.map((child) => nodesMap[child.id]);
```